src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,212 @@
+            break;
+        default:
+            break;
+        }
+        recordedTask.get(Parser.getOrderAdded() - 1).setTaskName(taskCommandArr[1] + exactDueDate); //formats the output statement as desired
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.TASK_ADDER_AFFIRMATION);
+        recordedTask.get(Parser.getOrderAdded() - 1).printTaskListing();
+        if (Parser.getOrderAdded() == 1) {
+            System.out.println(Messages.TASK_ADDER_SINGULAR);
+        } else {
+            System.out.println(Messages.TASK_ADDER_PLURAL);
+        }
+        System.out.println(Messages.LINE_DIVIDER);
+        orderAdder();
+    }
+
+    private static void printEmptyTaskErrorMessage(String task) {
+        String statementPartOne = task.equals("event") ? Messages.EMPTY_TASK_ERROR_MESSAGE + "n ": Messages.EMPTY_TASK_ERROR_MESSAGE + " "; //an event vs a todo or deadline
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(statementPartOne + task + Messages.EMPTY_TASK_ERROR_MESSAGE_END);
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void printUnrecognisedCommandErrorMessage() {
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.UNRECOGNISED_COMMAND);
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void listOfTasksPrinter(ArrayList<Task> recordedTask) {
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.LIST_TASKS_MESSAGE);
+        for (int i = 0; i < Parser.getOrderAdded() - 1; i++) {
+            recordedTask.get(i).printEntireTaskList();
+        }
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void setTaskAsDone(String commandEntered, ArrayList<Task> recordedTask) {
+        try {
+            if (!commandEntered.substring(6).equals("")) {
+                throw new DukeException();
+            }
+        } catch (DukeException de) {
+            printUnrecognisedCommandErrorMessage();
+            return;
+        }
+        char taskNumToCheck = commandEntered.charAt(5);
+        int taskNumToChange = Character.getNumericValue(taskNumToCheck);
+        recordedTask.get(taskNumToChange - 1).setTaskStatus(true);
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.MARKED_TASKS_DONE_MESSAGE);
+        recordedTask.get(taskNumToChange - 1).markedAsDone();
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    private static void printByeMessage() {
+        System.out.println(Messages.LINE_DIVIDER);
+        System.out.println(Messages.BYE_MESSAGE);
+        System.out.println(Messages.LINE_DIVIDER);
+    }
+
+    protected static void deleteTask(String commandEntered, ArrayList<Task> recordedTask) {
+        try {
+            if (!commandEntered.substring(8).equals("")) {
```

Comment:
Most of the command methods use magic numbers such as this eg. substring(8). Perhaps try to value add to this number by saving it as a variable or a constant?